### PR TITLE
[MIEB] Add Stanford Cars i2i Retrieval

### PR DESCRIPTION
**Pull request by isaac-chung**
The results are likely different from the reported ones from literature as the images are not cropped. I think it's okay to leave this as is.

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

```
mteb run -m openai/clip-vit-base-patch32 --model_revision 3d74acf9a28c67741b2f4f2ea7635f0aaf6f0268 -t StanfordCarsI2IRetrieval
```

### CODE REVIEW

File: mteb/evaluation/evaluators/Image/Any2AnyRetrievalEvaluator.py
Comment (gowitheflow-1998) [inline mteb/evaluation/evaluators/Image/Any2AnyRetrievalEvaluator.py:1]: any reason for this? I think `queries["modality"][0]` is much more expensive when `queries` is large.

Comment (isaac-chung) [reply mteb/evaluation/evaluators/Image/Any2AnyRetrievalEvaluator.py:1]: Good catch, I forgot to change it back.

File: mteb/tasks/Image/Any2AnyRetrieval/eng/StanfordCarsI2IRetrieval.py
Comment (gowitheflow-1998) [inline mteb/tasks/Image/Any2AnyRetrieval/eng/StanfordCarsI2IRetrieval.py:1]: think `average_relevant_docs_per_query` should be ~40. I will merge and compute it later no worries.

Comment (isaac-chung) [reply mteb/tasks/Image/Any2AnyRetrieval/eng/StanfordCarsI2IRetrieval.py:1]: Thanks!

**Pull request merged**